chef/cookbooks/ceph/recipes/conf.rb

```diff
@@ -62,6 +62,9 @@
 # calculate pg_num based on documentation
 # http://ceph.com/docs/master/rados/operations/placement-groups/
 case num_osds
+when 0
+  Chef::Log.warn("Ceph recipe invoked but there are no OSDs!")
+  return
```

Comment:
I have also doubts if we really should return here, when pg_num will be 0 ceph.conf can be also generated. During next run when OSDs will be available ceph.conf will be regenerated once again.